File: java/Mahasiswa.java
public class Mahasiswa 
{
	//attribut//

	private String name;
	//attribut nama
	private String nim;
	//attribut nim
	private String prodi;
	//attribut prodi
	private String fakultas;
	//attribut fakultas


	//konstruktor//

	Mahasiswa() {
		// konstruktor kosong
	}

	Mahasiswa(String nama, String nim, String prodi, String fakultas) {
		// konstruktor langsung isi atribut
		this.name = nama;
		this.nim = nim;
		this.prodi = prodi;
		this.fakultas = fakultas;
	}



	//setter and getter//

	public void setNama(String nama) {
		// mengeset nilai nama Mahasiswa
		this.name = nama;
	}

	public String getNama() {
		//mengembalikan nilai nama mahasiswa
		return this.name;
	}

	public void setNim(String nim) {
		// mengeset nilai nim Mahasiswa
		this.nim = nim;
	}

	public String getNim() {
		//mengembalikan nilai nim mahasiswa
		return this.nim;
	}

	public void setProdi(String prodi) {
		// mengeset nilai prodi Mahasiswa
		this.prodi = prodi;
	}

	public String getProdi() {
		//mengembalikkan nilai prodi mahasiswa
		return this.prodi;
	}

	public void setFakultas(String fakultas) {
		// mengeset nilai fakultas Mahasiswa
		this.fakultas = fakultas;
	}

	public String getFakultas() {
		//mengembalikkan nilai fakultas mahasiswa
		return this.fakultas;
	}

	

}

File: c++/Mahasiswa.cpp
#include <iostream>
#include <string>



//Using standard namespace
using namespace std;

// Class declaration. For C++, first letter doesn't have to be
//capitalized, but it's better so we can distinguish it with the other types.
class Mahasiswa
{
	//private attributes
	private:
		string name;
		string nim;
		string prodi;
		string fakultas;

	//public methods
	public:

		/* constructor */

		//constructor, doesn't have any return type
		Mahasiswa(){
			//set to default value
			this->name = "";
			this->nim = "";
			this->prodi = "";
			this->fakultas = "";
		}

		//another constructor with parameter. This one will be called if
		//the object is intanced with the given parameter.
		Mahasiswa(string name, string nim, string prodi, string fakultas){
			//"this" object's attributes will be set with the given parameter attributes.
			this->name = name;
			this->nim = nim;
			this->prodi = prodi;
			this->fakultas = fakultas;
		}

		/* getter and setter */

		//get name
		string getName(){
			return this->name;
		}

		//set name
		void setName(string name){
			this->name = name;
		}

		//get nim
		string getNim(){
			return this->nim;
		}

		//set nim
		void setNim(string nim){
			this->nim = nim;
		}

		//get prodi
		string getProdi(){
			return this->prodi;
		}

		//set prodi
		void setProdi(string prodi){
			this->prodi = prodi;
		}

		//get fakultas
		string getFakultas(){
			return this->fakultas;
		}

		//set fakultas
		void setFakultas(string fakultas){
			this->fakultas = fakultas;
		}


		
		/* destructors */

		//default destructor
		~Mahasiswa(){
			
		}
	
};